### PR TITLE
working on api spec

### PR DESCRIPTION
**Pull request by pupapaik**

### CODE REVIEW

File: api.yml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    type: object
+    properties:
+      id:
+        type: "integer"
+        format: "int64"
+      name:
+        type: "string"
+      provisioner:
+        type: "integer"
+        format: "int64"
```
Comment (tomkukral) [inline api.yml]: Is provisioner really int64?

File: api.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          - "deployed"
+          - "pending"
+          - "failed"
+          - "deleted"
+  ClusterNode:
+    type: object
+    properties:
+      name:
+        type: string
+      # TODO: more properties
```
Comment (tomkukral) [inline api.yml]: I'd suggest to dont' add more properties, lets use Kuberneties native node.

File: api.yml [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  ClusterService:
+    type: object
+    properties:
+      name:
```
Comment (tomkukral) [inline api.yml]: Once more, lets use native Kubernetes services.

File: api.yml [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        type: string
+      # TODO: more properties
+  Provisioner:
+    type: object
+    properties:
+      id:
+        type: "integer"
```
Comment (tomkukral) [inline api.yml]: What does this id means?

File: api.yml [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  - http
+
+basePath: /api/v1
+
+definitions:
+  Cluster:
+    type: object
+    properties:
+      id:
+        type: "integer"
```
Comment (Frostman) [inline api.yml]: usually it's better not to use int as id. some uuid (string) or human-readable (string as well) will be more flexible.

File: api.yml (hunk 1 above)
Comment (fpytloun) [reply api.yml]: It's id of linked provisioner object

File: api.yml (hunk 4 above)
Comment (fpytloun) [reply api.yml]: unique identifier of that object, or we can use names as unique identifiers and don't use ids at all

Comment (Frostman) [reply api.yml]: ID and name aren't the same - if you delete object with some name and than create new one with same name, the ID should be changed, so, you can differentiate the objects.

File: api.yml (hunk 5 above)
Comment (tomkukral) [reply api.yml]: uuid is a good idea, thanks!

**Pull request merged**